tools/generate-examples.rb

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'pathname'
+
+GraalVM = Struct.new(:url, :dir) do
+  def name
+    dir
+  end
+
+  def tarball
+    @tarball ||= File.basename(url)
+  end
+end
+
+GraalVM_21_2_0 = GraalVM.new('https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-21.2.0/graalvm-ce-java11-darwin-amd64-21.2.0.tar.gz', 'graalvm-ce-java11-21.2.0')
+GraalVM_22_3_1 = GraalVM.new('https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.3.1/graalvm-ce-java17-darwin-amd64-22.3.1.tar.gz', 'graalvm-ce-java17-22.3.1')
+
+GRAAL_VMS = [
+  GraalVM_22_3_1,
+  GraalVM_21_2_0
+]
+
+def run_java(bin, dump_level, *args)
+  FileUtils.rm_rf('./graal_dumps')
+
+  pipe = IO.popen(["#{bin}/java", \
+                   '-XX:+PrintCompilation', \
+                   '-Dgraal.PrintGraphWithSchedule=true', \
+                   "-Dgraal.Dump=:#{dump_level}"] + args)
+
+  yield pipe
+
+  sleep 3
+  Process.kill 'KILL', pipe.pid
+  Process.wait pipe.pid
+end
+
+def run_js(bin, *args)
+  FileUtils.rm_rf('graal_dumps')
+
+  pipe = IO.popen(["#{bin}/node", '--experimental-options', \
+                   '--engine.CompileOnly=fib', '--engine.TraceCompilation', \
+                   '--engine.MultiTier=false', '--engine.Inlining=false', \
+                   '--vm.Dgraal.Dump=Truffle:1'] + args, { err: %i[child out] })
+
+  yield pipe
+
+  sleep 3
+  Process.kill 'KILL', pipe.pid
+  Process.wait pipe.pid
+end
+
+def run_ruby(bin, *args)
+  FileUtils.rm_rf('./graal_dumps')
+
+  pipe = IO.popen(["#{bin}/ruby", '--jvm', '--experimental-options', \
+                   '--engine.TraceCompilation', \
+                   '--engine.NodeSourcePositions', '--engine.MultiTier=false', \
+                   '--engine.Inlining=false', '--vm.Dgraal.Dump=Truffle:1'] + args, { err: %i[child out] })
+
+  yield pipe
+
+  sleep 3
+  Process.kill 'KILL', pipe.pid
+  Process.wait pipe.pid
+end
+
+def process_examples(graalvm, language, pattern)
+  Dir.mkdir language unless Dir.exist?(language)
```

Comment:
Since we already have `fileutils` we could replace all of these `Dir.mkdir(x) unless Dir.exist?(x)` with `FileUtils.mkdir_p(x)`